sdk/eventhub/Microsoft.Azure.EventHubs/assets/live-tests-azure-setup.ps1

```diff
@@ -0,0 +1,325 @@
+
+if (!$validLocations.Contains($AzureRegion))
+{
+    Write-Error "The Azure region must be one of: `n$($validLocations.Keys -join ", ")`n`n" 
+    exit -1
+}
+
+# Capture the subscription.  The cmdlet will error if there was no subscription, 
+# so no need to validate.
+
+Write-Host ""
+Write-Host "Working:"
+Write-Host "`t...Requesting subscription"
+$subscription = (Get-AzSubscription -SubscriptionName "$($SubscriptionName)" -ErrorAction SilentlyContinue)
+
+if ($subscription -eq $null)
+{
+    Write-Error "Unable to locate the requested Azure subscription: $($SubscriptionName)"
+    exit -1
+}
+
+Set-AzContext -SubscriptionId "$($subscription.Id)" -Scope "Process" | Out-Null
+
+# Create the resource group, if needed.
+
+Write-Host "`t...Requesting resource group"
+
+$createResourceGroup = $false
+$resourceGroup = (Get-AzResourceGroup -ResourceGroupName "$($ResourceGroupName)" -ErrorAction SilentlyContinue)
+
+if ($resourceGroup -eq $null)
+{
+    $createResourceGroup = $true
+}
+
+if ($createResourceGroup)
+{
+    Write-Host "`t...Creaating new resource group"
+    $resourceGroup = (New-AzResourceGroup -Name "$($ResourceGroupName)" -Location "$($AzureRegion)")
+}
+
+if ($resourceGroup -eq $null)
+{
+    Write-Error "Unable to locate or create the resource goup: $($ResourceGroupName)"
+    exit -1
+}
+
+# At this point, we may have created a resource, so be safe and allow for removing any
+# resoruces created should the script fail.
+
+try 
+{
+    # Create the service prinicpal and grant contributor access for management in the resource group.
+
+    Write-Host "`t...Creating new service principal"
+    Start-Sleep 1
+
+    $credentials = New-Object Microsoft.Azure.Commands.ActiveDirectory.PSADPasswordCredential -Property @{StartDate=Get-Date; EndDate=Get-Date -Year 2099; Password="$(GenerateRandomPassword)"}            
+    $principal = (New-AzADServicePrincipal -DisplayName "$($ServicePrincipalName)" -PasswordCredential $credentials)
+
+    if ($principal -eq $null)
+    {
+        Write-Error "Unable to create the service principal: $($ServicePrincipalName)"
+        TearDownResources $createResourceGroup
+        exit -1
+    }
+    
+    Write-Host "`t...Assigning permissions (this will take a moment)"
+    Start-Sleep 60
+
+    # The propagation of the identity is non-deterministic.  Attempt to retry once after waiting for another minute if
+    # the initial attempt fails.
+
+    try 
+    {
+        New-AzRoleAssignment -ApplicationId "$($principal.ApplicationId)" -RoleDefinitionName "Contributor" -ResourceGroupName "$($ResourceGroupName)" | Out-Null
+    }
+    catch 
+    {
+        Write-Host "`t...Still waiting for identity propagation (this will take a moment)"
+        Start-Sleep 60
+        New-AzRoleAssignment -ApplicationId "$($principal.ApplicationId)" -RoleDefinitionName "Contributor" -ResourceGroupName "$($ResourceGroupName)" | Out-Null
+    }    
+
+    # Write the environment variables

```

Comment:
Super nit: maybe we ending the comment with a period to keep it consistent with other comments in the file?